compiler/codegen_llvm/lib/lumen/EIR/IR/EIROps.cpp

```diff
@@ -2414,8 +2414,10 @@ static LogicalResult verify(MallocOp op) {
         if (!termType.isBoxable())
             return op.emitOpError("cannot malloc an unboxable term type");
 
-        Optional<Value> arityVal = op.arity();
-        if (arityVal.hasValue()) {
+        // Optional<arity> in `arguments` is `Value()`, that is `Value(nullptr)` if not given and not an
```

Comment:
Can we revert any changes to the C++ code in this PR? I've made way too many changes locally for me to be able to merge them cleanly after this. A massive amount has changed. As an alternative, you could open a separate PR that contains these changes, and I can manually apply any changes that still carry over, but there will be no way for me to sanely merge against this.

Reply:
Not this one, but the one below about the `op.size()` is required to get the `lumen` compiler tests to pass at the same level as develop.  With @hansihe's changes to EIR to fix the binary code to not push a single byte at a time, but the whole binary, line 2747-2748 triggered an `assert`.

If I remove this changes the `compiler` workflows will be broken once this merges.